src/io/parsers/MOL2Parser.js

```diff
@@ -0,0 +1,265 @@
+import * as THREE from 'three';
+import Parser from './Parser';
+import chem from '../../chem';
+import MOL2Stream from './MOL2Stream';
+
+const {
+  Complex,
+  Element,
+  Bond,
+  Molecule,
+} = chem;
+
+const orderMap = {
+  un: 0,
+  1: 1,
+  2: 2,
+  3: 3,
+  ar: 1,
+  am: 1,
+  nc: 0,
+  du: 1,
+};
+const typeMap = {
+  un: Bond.BondType.UNKNOWN, // unknown (cannot be determined from the parameter tables)
+  1: Bond.BondType.COVALENT, // single
+  2: Bond.BondType.COVALENT, // double
+  3: Bond.BondType.COVALENT, // triple
+  ar: Bond.BondType.AROMATIC, // aromatic
+  am: Bond.BondType.COVALENT, // amide
+  nc: Bond.BondType.UNKNOWN, // not connected
+  du: Bond.BondType.COVALENT, // dummy
+};
+
+const resNumberRegex = /\d+$/;
+const spacesRegex = /\s+/;
+
+function splitToFields(str) {
+  return str.trim().split(spacesRegex);
+}
+/* There is no jsdoc documentation because of eslint corrections:
+ * not all Parser methods are implemented
+ */
+
+class MOL2Parser extends Parser {
+  constructor(data, options) {
+    super(data, options);
+
+    this._complex = null;
+    this._chain = null;
+    this._residue = null;
+    this._compoundIndx = -1;
+
+    this._molecules = [];
+    this._molecule = null;
+
+    this._serialAtomMap = {};
+
+    this._options.fileType = 'mol2';
+  }
+
+  _parseMolecule(stream) {
+    stream.getHeaderString('MOLECULE');
+
+    const { metadata } = this._complex;
+    metadata.name = stream.getNextString();
+    metadata.format = 'mol2';
+
+    this._molecule = { _index: '', _chains: [] };
+    this._molecule._index = this._compoundIndx + 1;
+    this._molecules.push(this._molecule);
+  }
+
+  /* Atom format description:
+   * atomId atomName x y z element [resSeq [resName [charge [statusBit]]]]
+   * statusBits is the internal SYBYL status bits associated with the atom.
+   * These should never be set by the user.
+   * Source: http://chemyang.ccnu.edu.cn/ccb/server/AIMMS/mol2.pdf
+   */
+  _parseAtoms(stream, atomsNum) {
+    let curStr = stream.getHeaderString('ATOM');
+
+    for (let i = 0; i < atomsNum; i++) {
+      curStr = stream.getNextString();
+      const parsedStr = splitToFields(curStr);
+
+      if (parsedStr.length < 6) {
+        throw new Error('MOL2 parsing error: Not enough information to create atom!');
+      }
+      const atomId = parseInt(parsedStr[0], 10);
+      const atomName = parsedStr[1];
+
+      const x = parseFloat(parsedStr[2]);
+      const y = parseFloat(parsedStr[3]);
+      const z = parseFloat(parsedStr[4]);
+
+      const element = parsedStr[5].split('.')[0].toUpperCase();
+
+      let charge = 0;
+      if (parsedStr.length >= 9) {
+        charge = parseFloat(parsedStr[8]) | 0;
```

Comment:
I'm afraid it is a double "or":
```suggestion
        charge = parseFloat(parsedStr[8]) || 0.0;
```

Reply:
Done